### ISSUE TITLE
[3.x][GLES3] ReflectionProbe not working under a viewport node with "own world" set to be true

### ISSUE THREAD

**Issue by Gawehold** [godotengine/godot#60950]
### Godot version

3.5.beta1.mono and 3.5.beta5.mono

### System information

Windows 10 64-bit, GLES3, GTX1060 6GB

### Issue description

I am trying to use a separated viewport to display the 3D world to have better controls of 3D rendering resolution without messing the 2D resolution. However, ReflectionProbe node doesn't work when it is placed under a viewport node and the viewport's "own world" setting set to be true. I want to use "own world" because it appears to have better performance.
![reflection-probe-viewport-bug-own-world-off](https://user-images.githubusercontent.com/48320629/167917562-70ca05f8-2510-4515-889a-e9be30753782.png)
![reflection-probe-viewport-bug-own-world-on](https://user-images.githubusercontent.com/48320629/167917568-6b9d2665-c7fd-4c6f-8f32-2770926ca1d5.png)

### Steps to reproduce

1. Open the following minimal reproduction project.
2. Toggle on and off the "own world" setting of the viewport node.
3. The reflection probe isn't working when the viewport's "own world" setting set to be true.

### Minimal reproduction project

[ReflectionProbeBug.zip](https://github.com/godotengine/godot/files/8672330/ReflectionProbeBug.zip)

**Comment by Gawehold** [comment]
Just to add that for reference, I tested it in Godot 4.0 alpha 7 and it seems to be working fine.

**Comment by Calinou** [comment]
Can you reproduce this after switching to the GLES2 renderer?

**Comment by Gawehold** [comment]
> Can you reproduce this after switching to the GLES2 renderer?

In GLES2, reflection probe doesn't even look the same as in GLES3. In my example project, it only makes the objects slightly darker which is very different to the first image I showed, but I guess it is just how it works or at least it is another topic.

At least how the reflection probe behave doesn't affected by the viewport's "own world" setting in GLES2, so I would say it is working in GLES2, so maybe it is a GLES3-specific problem.

**Comment by Gawehold** [comment]
I think it might be related the default reflection atlas size of the viewport.
I added this line of code for the viewport and it seems to be working now:
`VisualServer.scenario_set_reflection_atlas_size(self.find_world().scenario, 2048, 8);`

**Comment by Calinou** [comment]
> I think it might be related the default reflection atlas size of the viewport. I added this line of code for the viewport and it seems to be working now: `VisualServer.scenario_set_reflection_atlas_size(self.find_world().scenario, 2048, 8);`

Indeed, secondary viewports do not have a reflection atlas by default for performance reasons. The same applies to the shadow atlas size (it defaults to 0 in Viewport).

**Comment by Gawehold** [comment]
> > I think it might be related the default reflection atlas size of the viewport. I added this line of code for the viewport and it seems to be working now: `VisualServer.scenario_set_reflection_atlas_size(self.find_world().scenario, 2048, 8);`
> 
> Indeed, secondary viewports do not have a reflection atlas by default for performance reasons. The same applies to the shadow atlas size (it defaults to 0 in Viewport).
> 
> There should probably be a high-level property to set the reflection atlas size on a World3D though.

That makes sense to me. It would be more clear if there are settings in the editor, just like the shadow atlas ones.

What do you think about adding options to follow main viewport's settings though (separated options for shadow size, reflection size, etc)? It would be convenient for some tasks, and can potentially avoid troubles for the user who are not familiar with these settings ([example](https://www.reddit.com/r/godot/comments/si9a1k/shadows_arent_appearing_in_my_viewports_when_i/)), because they can just turn on and off the options to see if it works for them.

**Comment by Calinou** [comment]
> What do you think about adding options to follow main viewport's settings though (separated options for shadow size, reflection size, etc)? It would be convenient for some tasks, and can potentially avoid troubles for the user who are not familiar with these settings ([example](https://www.reddit.com/r/godot/comments/si9a1k/shadows_arent_appearing_in_my_viewports_when_i/)), because they can just turn on and off the options to see if it works for them.

This makes sense, but it would have to remain disabled by default for performance reasons. Also, any change to the root Viewport settings should automatically be reflected on other Viewports using the Inherit property value.

**Comment by Gawehold** [comment]
> This makes sense, but it would have to remain disabled by default for performance reasons. Also, any change to the root Viewport settings should automatically be reflected on other Viewports using the Inherit property value.
> 
> The same applies to other "quality" settings on Viewport such as MSAA, screen-space AA and debanding.

Sounds good to me, but may I ask that how can it work for the reflections atlas size? If I didn't misunderstand it, reflections atlas size is an attribute of the world (or scenario?) instead of the viewport, then it seems to be a bit tricky to add the "follow" option or size settings to the viewport.

What about automatically adding a new world to the "world" slot when the slot is empty and the "own world" setting is enabled + adding options under the world resource for the reflection atlas size and/or following the world from the main viewport?
